kube_apiserver_metrics/datadog_checks/kube_apiserver_metrics/kube_apiserver_metrics.py

```diff
@@ -119,6 +121,8 @@ def __init__(self, name, init_config, instances=None):
             # For Kubernetes >= 1.24
             # https://github.com/kubernetes/kubernetes/pull/107171
             'apiserver_admission_webhook_fail_open_count': self.apiserver_admission_webhook_fail_open_count,
+            # https://github.com/kubernetes/kubernetes/pull/103162
+            'apiserver_admission_webhook_request_total': self.apiserver_admission_webhook_request_total,
```

Comment:
Actually, unit tests are failing:
```
tests/test_kube_apiserver_metrics_1_23.py:97: in test_check
    aggregator.assert_metric(metric_to_assert)
../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:372: in assert_metric
    self._assert(condition, msg=msg, expected_stub=expected_metric, submitted_elements=self._metrics)
../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:414: in _assert
    assert condition, new_msg
E   AssertionError: Needed at least 1 candidates for 'kube_apiserver.apiserver_admission_webhook_request_total', got 0
E     Expected:
E             MetricStub(name='kube_apiserver.apiserver_admission_webhook_request_total', type=None, value=None, tags=None, hostname=None, device=None, flush_first_value=None)
E     Similar submitted:
E     Score   Most similar
E     0.90    MetricStub(name='kube_apiserver.admission_webhook_request_total', type=3, value=18.0, tags=['code:200', 'custom:tag', 'name:datadog.webhook.config', 'operation:CREATE', 'rejected:false', 'type:admit'], hostname='', device=None, flush_first_value=True)
E     0.90    MetricStub(name='kube_apiserver.admission_webhook_request_total', type=3, value=18.0, tags=['code:200', 'custom:tag', 'name:datadog.webhook.config', 'operation:CREATE', 'rejected:false', 'type:admit'], hostname='', device=None, flush_first_value=False)
E     0.90    MetricStub(name='kube_apiserver.admission_webhook_request_total', type=3, value=17.0, tags=['code:200', 'custom:tag', 'name:datadog.webhook.tags', 'operation:CREATE', 'rejected:false', 'type:admit'], hostname='', device=None, flush_first_value=True)
E     0.90    MetricStub(name='kube_apiserver.admission_webhook_request_total', type=3, value=17.0, tags=['code:200', 'custom:tag', 'name:datadog.webhook.tags', 'operation:CREATE', 'rejected:false', 'type:admit'], hostname='', device=None, flush_first_value=False)
E     0.90    MetricStub(name='kube_apiserver.admission_webhook_request_total', type=3, value=17.0, tags=['code:200', 'custom:tag', 'name:datadog.webhook.auto.instrumentation', 'operation:CREATE', 'rejected:false', 'type:admit'], hostname='', device=None, flush_first_value=True)
E     0.90    MetricStub(name='kube_apiserver.admission_webhook_request_total', type=3, value=17.0, tags=['code:200', 'custom:tag', 'name:datadog.webhook.auto.instrumentation', 'operation:CREATE', 'rejected:false', 'type:admit'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=4567.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:coordination.k8s.io', 'resource:leases', 'scope:resource', 'subresource:', 'verb:PUT', 'version:v1'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=4567.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:coordination.k8s.io', 'resource:leases', 'scope:resource', 'subresource:', 'verb:PUT', 'version:v1'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=4180.0, tags=['code:200', 'component:', 'custom:tag', 'dry_run:', 'group:', 'resource:', 'scope:', 'subresource:/readyz', 'verb:GET', 'version:'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=4180.0, tags=['code:200', 'component:', 'custom:tag', 'dry_run:', 'group:', 'resource:', 'scope:', 'subresource:/readyz', 'verb:GET', 'version:'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=4157.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:coordination.k8s.io', 'resource:leases', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=4157.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:coordination.k8s.io', 'resource:leases', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=2487.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=2487.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E     0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=2083.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:PUT', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   assert False
```

This is because you have defined the metric in BOTH the `METRICS` list above and in `self.metric_transformers` here. It seems like this is the expected path that you were going with, so you should remove the entry from `METRICS` on line 77-78

Reply:
Done! But the tests are still failing. I'm trying to figure out what's happening.

Reply:
```
tests/test_kube_apiserver_metrics_1_26.py::TestKubeAPIServerMetrics::test_check FAILED [ 57%]
tests/test_kube_apiserver_metrics_1_27.py::TestKubeAPIServerMetrics::test_check FAILED [ 63%]
tests/test_kube_apiserver_metrics_1_28.py::TestKubeAPIServerMetrics::test_check FAILED [ 68%]
```

Those are the test suites still failing, all with the same output (so only copying one):
```
tests/test_kube_apiserver_metrics_1_28.py:99: in test_check
    aggregator.assert_metric(metric_to_assert)
../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:372: in assert_metric
    self._assert(condition, msg=msg, expected_stub=expected_metric, submitted_elements=self._metrics)
../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:414: in _assert
    assert condition, new_msg
E   AssertionError: Needed at least 1 candidates for 'kube_apiserver.apiserver_admission_webhook_request_total', got 0
E   Expected:
E           MetricStub(name='kube_apiserver.apiserver_admission_webhook_request_total', type=None, value=None, tags=None, hostname=None, device=None, flush_first_value=None)
E   Similar submitted:
E   Score   Most similar
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=48171.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:configmaps', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=48171.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:configmaps', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=18656.0, tags=['code:200', 'component:', 'custom:tag', 'dry_run:', 'group:', 'resource:', 'scope:', 'subresource:/readyz', 'verb:GET', 'version:'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=18656.0, tags=['code:200', 'component:', 'custom:tag', 'dry_run:', 'group:', 'resource:', 'scope:', 'subresource:/readyz', 'verb:GET', 'version:'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=15137.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:nodes', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=15137.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:nodes', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=14863.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=14863.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:GET', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=11302.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:pods', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=11302.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:pods', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=11174.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:batch', 'resource:jobs', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=11174.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:batch', 'resource:jobs', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=9300.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:PUT', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=9300.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:endpoints', 'scope:resource', 'subresource:', 'verb:PUT', 'version:v1'], hostname='', device=None, flush_first_value=False)
E   0.81    MetricStub(name='kube_apiserver.apiserver_request_total', type=0, value=7749.0, tags=['code:200', 'component:apiserver', 'custom:tag', 'dry_run:', 'group:', 'resource:services', 'scope:cluster', 'subresource:', 'verb:WATCH', 'version:v1'], hostname='', device=None, flush_first_value=False)
```

Basically, the change that you made to the kube_apiserver_metrics/tests/fixtures/metrics_1.23.0.txt fixture file needs to also be added to the fixtures used for [1.26](https://github.com/jwillker/integrations-core/blob/add/admission-webhook-request/kube_apiserver_metrics/tests/fixtures/metrics_1.26.0.txt), [1.27](https://github.com/jwillker/integrations-core/blob/add/admission-webhook-request/kube_apiserver_metrics/tests/fixtures/metrics_1.27.0.txt), and [1.28](https://github.com/jwillker/integrations-core/blob/add/admission-webhook-request/kube_apiserver_metrics/tests/fixtures/metrics_1.28.0.txt)

What is interesting, however, is that kube_apiserver_metrics/tests/test_kube_apiserver_metrics_1_24.py and kube_apiserver_metrics/tests/test_kube_apiserver_metrics_1_25.py are passing, and they too have their own test fixtures ([1.24](https://github.com/jwillker/integrations-core/blob/add/admission-webhook-request/kube_apiserver_metrics/tests/fixtures/metrics_1.24.0.txt), [1.25](https://github.com/jwillker/integrations-core/blob/add/admission-webhook-request/kube_apiserver_metrics/tests/fixtures/metrics_1.25.3.txt)) so I would assume those would also need to be updated, but I would expect a failure because they haven't been updated

Reply:
@sblumenthal Fixed. All tests are ok